src/test/ui/parser/closure-return-syntax.stderr

```diff
@@ -1,8 +1,11 @@
-error: expected one of `!`, `(`, `+`, `::`, `<`, or `{`, found `22`
+error: expected `{`, found `22`
   --> $DIR/closure-return-syntax.rs:5:23
    |
 LL |     let x = || -> i32 22;
-   |                       ^^ expected one of `!`, `(`, `+`, `::`, `<`, or `{`
+   |                       ^^-
+   |                       |
+   |                       expected `{`
+   |                       help: try placing this code inside a block: `{ 22; }`
```

Comment:
Haven't gotten to the code yet, but would it be possible to identify when you're parsing a closure? My expectation would be that the suggestion here should have been `{ 22 };`, otherwise we're suggesting invalid code.

Reply:
Found a good way to deal with closures and other cases of "there's something before the statement on the same line".